gratipay/billing/payday.py

```diff
@@ -706,6 +707,57 @@ def end(self):
         """, default=NoPayday).replace(tzinfo=aspen.utils.utc)
 
 
+    def notify_participants(self):
+        ts_start, ts_end = self.ts_start, self.ts_end
+        exchanges = self.db.all("""
+            SELECT amount, fee, note, status, p.*::participants AS participant
+              FROM exchanges e
+              JOIN participants p ON e.participant = p.username
+             WHERE "timestamp" >= %(ts_start)s
+               AND "timestamp" < %(ts_end)s
+               AND amount > 0
+               AND p.notify_charge > 0
+        """, locals())
+        for e in exchanges:
+            if e.status not in ('failed', 'succeeded'):
+                log('exchange %s has an unexpected status: %s' % (e.id, e.status))
+                continue
+            i = 1 if e.status == 'failed' else 2
+            p = e.participant
+            if p.notify_charge & i == 0:
```

Comment:
I find this code difficult to read. You mentioned ([IRC](https://botbot.me/freenode/gratipay/2015-04-06/?msg=35923796&page=1)) wanting to refactor how we're doing notifications. Does this point us in that direction? Are you thinking we'll fold all notifications together into one column that we'll work with via bitmasks? Are you expecting that we'll use bitwise operators with ints? I'd find it easier to comprehend bitwise code if it used `0b00` (Python) and `B'00'` (Postgres). It'd be even easier if we used constants (e.g.,  `NOTIFY_CHARGE_FAILURE`).